tests/test_simulation_execution_options.py

```diff
@@ -86,7 +86,34 @@ def test_binary_name_extension_on_windows(self):
             options = SimulationExecutionOptions()
             self.assertTrue(options.binary_name.endswith(".exe"))
 
-    def test_get_options_string(self):
+    @patch("kwave.options.simulation_execution_options.PLATFORM", "darwin")
+    def test_get_options_string_darwin(self):
+        """Test the get_options_string method with a mock sensor."""
+        options = self.default_options
+        options.device_num = 1
+        options.num_threads = os.cpu_count()
+        options.verbose_level = 2
+
+        options_string = options.get_options_string(self.mock_sensor)
+        expected_substrings = [" -g 1", f" -t {os.cpu_count()}", " --verbose 2", " --p_raw", " --u_max", " -s 10"]
+        for substring in expected_substrings:
+            self.assertIn(substring, options_string)
+
+    @patch("kwave.options.simulation_execution_options.PLATFORM", "windows")
+    def test_get_options_string_windows(self):
+        """Test the get_options_string method with a mock sensor."""
+        options = self.default_options
+        options.device_num = 1
+        options.num_threads = os.cpu_count()
+        options.verbose_level = 2
+
+        options_string = options.get_options_string(self.mock_sensor)
+        expected_substrings = [" -g 1", " --verbose 2", " --p_raw", " --u_max", " -s 10"]
+        for substring in expected_substrings:
+            self.assertIn(substring, options_string)
```

Comment:
Maybe additionally check that `-t` option is not present in the `options_string`?